gateway/cli/src/main.rs

```diff
@@ -53,14 +66,37 @@ pub enum Commands {
 async fn main() {
     let cli = Cli::parse();
     match cli.command {
+        Commands::GenerateConfig { mut out_dir } => {
+            // Recursively create config directory if it doesn't exist
+            std::fs::create_dir_all(&out_dir).expect("Failed to create config directory");
+            // Create config file
+            out_dir.push("gateway.config");
```

Comment:
Might be good to use `.json` extension since it's a JSON file?

Reply:
I'll revisit. Some things that stood out for me earlier were:
- This user specified config file serves the function of a `.env`, so that might be a better extension?
- Several binaries I've looked up have this nice pattern of autogenerating `.config` or `.conf` file into some known hidden dir that a user can look up and modify. They generally don't store info in a JSON-like structure!